Enrich 6.13.0 (#1887)
snowplow/documentation: docs/pipeline/enrichments/available-enrichments/referrer-parser-enrichment/index.md

This is particularly useful when looking for traffic from specific search engine providers or social networks.

> Since version 6.13.0 of Enrich, it also classifies traffic sources that identify themselves through a [`utm_source` query parameter](#identifying-referrers-from-utm_source) rather than a `Referer` header, which is how most AI chatbots send their traffic.

Reviewer: ```suggestion
Since version 6.13.0 of Enrich, it also classifies traffic sources that identify themselves through a [`utm_source` query parameter](#identifying-referrers-from-utm_source) rather than (or in addition to) a `Referer` header. An example would be ChatGPT setting `utm_source=chatgpt.com`.
```

:::

## Identifying referrers from `utm_source`

:::note[Availability]
This feature is available since version 6.13.0 of Enrich.
:::

> Not every traffic source sets a `Referer` header. Many — AI chatbots in particular — instead append a `utm_source` query parameter to the links they send, for example:

Reviewer: ```suggestion
The `Referer` header is not set when users share or copy and paste the link, rather than click it directly. To circumvent this limitation and make attribution more reliable, some sites also append a `utm_source` query parameter. For example:
```

```
https://www.example.com/product?utm_source=chatgpt.com
```

> 

Reviewer: Note that an event might contain both a `Referer` header value _and_ a `utm_source` value. These values might not match. For example, if person A shared the above link with person B via Instagram messages, and person B clicked the link, `Referer` would be `instagram.com`. In these cases, you should rely on the `utm_source` value for first-touch attribution.

Reviewer: Sorry, this was meant as a `suggestion` to be added directly to the text. I clicked the wrong button. Could you add instead of the lines you added below?

Author: Sorry I missed that! I will remove https://github.com/snowplow/documentation/pull/1887/changes/caa973d78d1de5af31be3fac63252f967b51345d then.

Author: I updated.

> Because there's no referrer URL, the `refr_*` fields stay empty and this traffic is indistinguishable from direct traffic. To close that gap, the enrichment also matches the `utm_source` parameter of the page URL against the `utm_sources` entries in the referer database, and attaches a `utm_referrer` entity to the event when it recognizes the value.

Reviewer: ```suggestion
The enrichment parses both the `Referer` header and the `utm_source` parameter and preserves both pieces of information. The header, if present, populates the `refr_*` fields of the event. If a referrer is detected through the `utm_source` parameter, the enrichment adds a `utm_referrer` entity to the event.
```

The hosted database includes `utm_sources` entries for the major chatbot providers — ChatGPT, Claude.ai, Google Gemini, Microsoft Copilot, META.ai, Mistral.ai, Perplexity.ai, Character.AI and Poe — so this works out of the box, as long as you use the actively-maintained database file (`referers-5.3`). You can recognize additional sources by adding `utm_sources` to your [custom referrer mappings](#custom-referrer-mappings).

### Output entity

On a match, the enrichment attaches a [`utm_referrer`](https://github.com/snowplow/iglu-central/tree/master/schemas/com.snowplowanalytics.snowplow/utm_referrer/jsonschema/1-0-0) entity to `derived_contexts`:

```json
{
  "schema": "iglu:com.snowplowanalytics.snowplow/utm_referrer/jsonschema/1-0-0",
  "data": {
    "source": "ChatGPT",
    "medium": "chatbot"
  }
}
```

> | Field    | Description                                                                           |

Reviewer: For consistency, could you use the same widget that Miranda added to other enrichments? For example, see IP lookup and what it does with ASN.

| -------- | ------------------------------------------------------------------------------------- |
| `source` | The name of the matched source e.g., `ChatGPT`. Equivalent to `refr_source`.           |
| `medium` | The medium of the matched source e.g., `chatbot`. Equivalent to `refr_medium`.         |

> | `term`   | The search term. Always absent for `utm_source` matches. Equivalent to `refr_term`.    |

Reviewer: ```suggestion
| `term`   | The search term. Currently, always absent for `utm_source` matches. Equivalent to `refr_term`.    |
```

### How it relates to referrer URL parsing

The `utm_source` classification is deliberately kept separate from the `Referer` header parsing described above:

* It **doesn't populate** `refr_medium`, `refr_source`, or `refr_term`. Those fields continue to reflect the referrer URL only, so existing data models and queries are unaffected.

> * It's **additive**. If an event has both a recognized referrer URL and a recognized `utm_source`, you get the `refr_*` fields *and* the `utm_referrer` entity. The two can legitimately disagree — for instance, a chatbot link opened from an email client — and keeping them apart lets you decide which signal to trust.

Reviewer: ```suggestion
* It's **additive**. If an event has both a recognized referrer URL and a recognized `utm_source`, you get the `refr_*` fields *and* the `utm_referrer` entity.
```
Moved this explanation up
